carbons/carbons.go

```diff
@@ -63,3 +66,45 @@ func DisableIQ(ctx context.Context, s *xmpp.Session, iq stanza.IQ) error {
 	)), &v)
 	return err
 }
+
+type Received struct {
+	XMLName xml.Name `xml:"urn:xmpp:carbons:2 received"`
+}
+
+func (received Received) Wrap(r xml.TokenReader) xml.TokenReader {
+	return xmlstream.Wrap(
+		r,
+		xml.StartElement{Name: xml.Name{Space: NS, Local: "received"}},
+	)
+}
+
+func WrapReceived(received time.Time, r xml.TokenReader) xml.TokenReader {
```

Comment:
All exported identifiers need doc comments; for an example of the format see [`Forwarded.Wrap`](https://pkg.go.dev/mellium.im/xmpp/forward#Forwarded.Wrap).

Reply:
I also forgot that there can be other attributes on the delay element, oops. This should probably take a [`delay.Delay`](https://pkg.go.dev/mellium.im/xmpp/delay#Delay) instead. Sorry about that!

Reply:
In that one there is a separate way to provide a full delay: https://pkg.go.dev/mellium.im/xmpp@main/forward#Forwarded

We can consider updating that later though, I'm not sure why I ended up doing it this way. It doesn't need to be part of the carbons work.